test/TestPlasma.js

```diff
@@ -0,0 +1,278 @@
+const ethjsUtil = require('ethereumjs-util');
+const secp256k1 = require('secp256k1');
+const crypto = require('crypto');
+const BigNumber = require('bignumber.js');
+
+const Transaction = require('../js/lib/Transaction');
+const SparseMerkleTree = require('../js/lib/SparseMerkleTree');
+
+const { sha3, sha256, bufferToHex } = require('ethereumjs-util');
+
+const Plasma = artifacts.require('./Plasma.sol');
+
+const privKey = new Buffer('c87509a1c067bbde78beb793e6fa76530b6382a4c0241e5e4a9ec0a0f44dc0d3', 'hex');
+
+contract('Plasma', async ([owner]) => {
+  it('check balance', async function () {
+    assert.isTrue((await web3.eth.getBalance(owner) > 1.0));
+  });
+
+  let plasma;
+  it('create contract', async function () {
+    plasma = await Plasma.new(-1);
+  });
+
+  it('has an owner', async function () {
+    assert.equal(await plasma.owner(), owner);
+  });
+
+  it('set operator', async function () {
+    await plasma.setOperator(owner, true);
+  });
+
+  it('is operator', async function () {
+    assert.isTrue(await plasma.isOperator());
+  });
+
+  it('test merkle proof', async function () {
+    const merkleTree = new SparseMerkleTree({
+      5: sha3(Buffer.from("5")),
+      4: sha3(Buffer.from("4")),
+      2: sha3(Buffer.from("2")),
+      1: sha3(Buffer.from("1"))
+    }, 4);
+
+    assert.isTrue(await plasma.verifyProof(merkleTree.getHexProofForIndex(0), merkleTree.getHexRoot(), "0x0", 0));
+    assert.isTrue(await plasma.verifyProof(merkleTree.getHexProofForIndex(4), merkleTree.getHexRoot(), SparseMerkleTree.bufArrToHex([sha3(Buffer.from("4"))]), 4));
+    assert.isTrue(await plasma.verifyProof(merkleTree.getHexProofForIndex(7), merkleTree.getHexRoot(), "0x0", 7));
+  });
+
+  let depositTransaction1;
+  it('deposit 1', async function () {
```

Comment:
seems like test `deposit 1, 2, 3` are all the same tests and you are just testing to see that the deposits have been called three times. perhaps what we want then is having all of the setup in a `beforeEach` and creating a new transaction three times and see that it was called three times